x/likenft/keeper/msg_server_buy_nft_test.go

```diff
@@ -97,6 +99,92 @@ func TestBuyNFTNormalRoyalty(t *testing.T) {
 	ctrl.Finish()
 }
 
+// normal royalty and full pay
+func TestBuyNFTNormalRoyaltyAndFullPay(t *testing.T) {
+	// Setup
+	ctrl := gomock.NewController(t)
+	accountKeeper := testutil.NewMockAccountKeeper(ctrl)
+	bankKeeper := testutil.NewMockBankKeeper(ctrl)
+	iscnKeeper := testutil.NewMockIscnKeeper(ctrl)
+	nftKeeper := testutil.NewMockNftKeeper(ctrl)
+	msgServer, goCtx, k := setupMsgServer(t, keeper.LikenftDependedKeepers{
+		AccountKeeper: accountKeeper,
+		BankKeeper:    bankKeeper,
+		IscnKeeper:    iscnKeeper,
+		NftKeeper:     nftKeeper,
+	})
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	ctx = ctx.WithBlockTime(time.Date(2022, 1, 1, 0, 0, 0, 0, time.UTC))
+	goCtx = sdk.WrapSDKContext(ctx)
+
+	// Data
+	creatorAddressBytes := []byte{1, 1, 1, 1, 0, 0, 0, 0}
+	sellerAddressBytes := []byte{0, 1, 0, 1, 0, 1, 0, 1}
+	sellerAddress, _ := sdk.Bech32ifyAddressBytes("like", sellerAddressBytes)
+	buyerAddressBytes := []byte{1, 0, 1, 0, 1, 0, 1, 0}
+	buyerAddress, _ := sdk.Bech32ifyAddressBytes("like", buyerAddressBytes)
+	classId := "likenft1abcdef"
+	nftId := "nft1"
+	price := uint64(123456)
+	expiration := time.Date(2022, 4, 1, 0, 0, 0, 0, time.UTC)
+	fullPayToRoyalty := true
+	royaltyBasisPoints := uint64(234)
+	finalPrice := uint64(200000)
+
+	// Seed listing
+	k.SetListing(ctx, types.ListingStoreRecord{
+		ClassId:          classId,
+		NftId:            nftId,
+		Seller:           sellerAddressBytes,
+		Price:            price,
+		Expiration:       expiration,
+		FullPayToRoyalty: fullPayToRoyalty,
+	})
+
+	// Seed royalty config
+	k.SetRoyaltyConfig(ctx, types.RoyaltyConfigByClass{
+		ClassId: classId,
+		RoyaltyConfig: types.RoyaltyConfig{
+			RateBasisPoints: royaltyBasisPoints,
+			Stakeholders: []types.RoyaltyStakeholder{
+				{
+					Account: creatorAddressBytes,
+					Weight:  uint64(1),
+				},
+			},
+		},
+	})
+	royaltyAmount := uint64(math.Floor(float64(finalPrice) / 10000 * float64(10000)))
```

Comment:
Note that it's not guaranteed that `royaltyAmount` equals `finalPrice`, and `netAmout` equals `0`  when `fullPayToRoyalty := true`. If we have multiple stakeholders, some of the allocations may get rounded down. Same as the below test and tests in `msg_server_sell_nft_test.go`.